examples/llava/minicpmv_wrapper.h

```diff
@@ -0,0 +1,49 @@
+#ifndef MINICPMV_H
+#define MINICPMV_H
+
+#include "common.h"
+#include "clip.h"
+#include "llava.h"
+#include "llama.h"
+
+#ifdef LLAMA_SHARED
+#    if defined(_WIN32) && !defined(__MINGW32__)
+#        ifdef LLAMA_BUILD
+#            define MINICPMV_API __declspec(dllexport)
+#        else
+#            define MINICPMV_API __declspec(dllimport)
+#        endif
+#    else
+#        define MINICPMV_API __attribute__ ((visibility ("default")))
+#    endif
+#else
+#    define MINICPMV_API
+#endif
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+struct minicpmv_context {
+    struct llama_context * ctx_llama = NULL;
+    struct llama_model * model = NULL;
+};
+
+MINICPMV_API struct llama_model * llava_init(gpt_params * params);
+MINICPMV_API struct minicpmv_context * llava_init_context(gpt_params * params, llama_model * model);
+MINICPMV_API void llava_free(struct minicpmv_context * ctx_llava);
+
+MINICPMV_API struct clip_ctx * clip_init_context(gpt_params * params);
+MINICPMV_API struct uhd_image_embed * minicpmv_image_embed(gpt_params * params, const std::string & fname);
+
+MINICPMV_API bool eval_tokens(struct llama_context * ctx_llama, std::vector<llama_token> tokens, int n_batch, int * n_past);
+MINICPMV_API bool eval_id(struct llama_context * ctx_llama, int id, int * n_past);
+MINICPMV_API bool eval_string(struct llama_context * ctx_llama, const char* str, int n_batch, int * n_past, bool add_bos);
+MINICPMV_API void process_image(struct minicpmv_context * ctx_llava,  std::vector<std::vector<struct llava_image_embed *>> image_embed_slices, gpt_params * params, int &n_past);
```

Comment:
Should not mix C and C++. Either convert to pure C-style API and use `minicmpv_` prefix for the function calls, or remove the `extern "C"`, `MINICPMV_API`, etc. stuff and keep this an internal C++ header

Reply:
OK, this modification is a little bit troublesome, but I will solve it as soon as possible.

Reply:
[6fd0937](https://github.com/ggerganov/llama.cpp/pull/7599/commits/6fd0937e9ff69370bac0aed97ac7bd64b23fd72c)

Do you think this modification is okay?